javalib/src/main/scala/java/util/Base64.scala

```diff
@@ -145,35 +156,34 @@ object Base64 {
       dst.position
     }
 
-    private def dstRequiredLength(src: Array[Byte], offset: Int, length: Int): Int = {
-      // scalastyle:off return
-      if (length == 0) return 0
-      // scalastyle:on return
-
+    private def dstRequiredLength(src: Array[Byte]): Int = {
       var validBytes = 0
-      var padding = 0
+
       if (ignoreInvalid) {
-        for (i <- offset until length) {
+        src.indices.foreach{ i =>
           if (table(src(i) & 0xff) >= 0)
             validBytes += 1
         }
       } else {
-        validBytes = length-offset
+        validBytes = src.length
         /* We check the end for padding and compute the length from there. This is ok, if the rest contains garbage
         we'll have written something before throwing but the spec says "If the input byte array is not in valid Base64
         encoding scheme then some bytes may have been written to the output byte array before IllegalArgumentException
         is thrown."*/
-        if (src(length-1) == '=') {
-          padding += 1
-          if (src(length-2) == '=')
-          padding += 1
+        if (src.length >= 1 && src.last == '=') {
+          validBytes -= 1
+          if (src.length >= 2 && src(src.length-2) == '=')
+            validBytes -= 1
         }
+        if(src.length >= 1 && validBytes == 0)
+          throw new IllegalArgumentException("Input byte array has wrong 4-byte ending unit")
       }
-      dstMaxLength(validBytes) - (if (validBytes % 4 == 0) padding else 4 - (validBytes % 4))
+      dstMaxLength(validBytes)
     }
 
-    /** compute the destination length solely based on the source length length without knowing about padding */
-    private def dstMaxLength(srcLength: Int): Int = (srcLength+3) / 4 * 3
+    /** compute the destination length solely based on the source length without knowing about padding */
+    private def dstMaxLength(srcLength: Int): Int =
+      (srcLength + 3) / 4 * 3 - (if (srcLength % 4 == 0) 0 else 4 - (srcLength % 4))
```

Comment:
I moved the `- (if (srcLength ....` stuff here but I see no way of getting rid of that since it won't work for lengths that are not multiples of 4.